pkg/networkservice/connectioncontext/mtu/common.go

```diff
@@ -35,6 +38,9 @@ const (
 )
 
 func setVPPL2MTU(ctx context.Context, conn *networkservice.Connection, vppConn api.Connection, isClient bool) error {
+	if vlan.ToMechanism(conn.GetMechanism()) != nil {
+		return nil
+	}
```

Comment:
Could you help me to understand this... I don't quite follow.

Reply:
Hello @edwarnicke,
The forwarder is the client on a VLAN remote mechanism connection. The mechanism/vlan/mtu chain client fetch the MTU of the sub-interface and set its value in ConnectionContext. The Connectioncontext client chain element should not set any MTU value since it's fetched from this interface and is already set (more than that in case of VLAN remote mechanism the API call will fail since the sub-interface is not a hw-interface).

I moved this statement to pkg/networkservice/connectioncontext/mtu/client.go.